services/dbaas/database.go

```diff
@@ -36,17 +34,15 @@ func (c *PsqlClient) GetDatabases(ctx context.Context, clusterId string) (pgsql.
 
 func SetDatabasePgSqlData(d *schema.ResourceData, database *pgsql.DatabaseResource) error {
 	resourceName := "PgSQL database"
-	d.SetId(*database.Id)
-	if database.Properties == nil {
-		return fmt.Errorf("expected properties in the response for the PgSql database with ID: %s, but received 'nil' instead", *database.Id)
-	}
-	if database.Properties.Name != nil {
-		if err := d.Set("name", *database.Properties.Name); err != nil {
+	d.SetId(database.Id)
+
+	if database.Properties.Name != "" {
```

Comment:
@cristiGuranIonos do we need this check here? I don't think so.

Reply:
@adeatcu-ionos not sure if this was the case when I left the comment and I didn't check properly, but we need this check here because `Name` is a pointer.

Reply:
@adeatcu-ionos I still don't know why we have these checks here but I will leave them as they are because maybe I am missing some strange edge case.